instrumentation/mongo/mongo-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/mongo/v4_0/Mongo4ReactiveClientTest.java

```diff
@@ -0,0 +1,305 @@
+        testing()
+            .runWithSpan(
+                "setup",
+                () -> {
+                  MongoDatabase db = client.getDatabase(dbName);
+                  CountDownLatch latch = new CountDownLatch(1);
+                  db.createCollection(collectionName)
+                      .subscribe(toSubscriber(o -> latch.countDown()));
+                  latch.await(30, TimeUnit.SECONDS);
+                  return db.getCollection(collectionName);
+                });
+    ignoreTracesAndClear(1);
+    return collection;
+  }
+
+  @Override
+  public int insert(MongoCollection<Document> collection) throws Exception {
+    CompletableFuture<Long> count = new CompletableFuture<>();
+    collection
+        .insertOne(new Document("password", "SECRET"))
+        .subscribe(
+            toSubscriber(
+                result ->
+                    collection
+                        .estimatedDocumentCount()
+                        .subscribe(toSubscriber(c -> count.complete(((Long) c))))));
+    return Math.toIntExact(count.get(30, TimeUnit.SECONDS));
+  }
+
+  @Override
+  public MongoCollection<Document> setupUpdate(String dbName, String collectionName)
+      throws InterruptedException {
+    MongoCollection<Document> collection =
+        testing()
+            .runWithSpan(
+                "setup",
+                () -> {
+                  MongoDatabase db = client.getDatabase(dbName);
+                  CountDownLatch latch1 = new CountDownLatch(1);
+                  db.createCollection(collectionName)
+                      .subscribe(toSubscriber(o -> latch1.countDown()));
+                  latch1.await(30, TimeUnit.SECONDS);
+                  MongoCollection<Document> coll = db.getCollection(collectionName);
+                  CountDownLatch latch2 = new CountDownLatch(1);
+                  coll.insertOne(new Document("password", "OLDPW"))
+                      .subscribe(toSubscriber(o -> latch2.countDown()));
+                  latch2.await(30, TimeUnit.SECONDS);
+                  return coll;
+                });
+    ignoreTracesAndClear(1);
+    return collection;
+  }
+
+  @Override
+  public int update(MongoCollection<Document> collection) throws Exception {
+    CompletableFuture<UpdateResult> result = new CompletableFuture<>();
+    CompletableFuture<Integer> count = new CompletableFuture<>();
+    collection
+        .updateOne(
+            new BsonDocument("password", new BsonString("OLDPW")),
+            new BsonDocument("$set", new BsonDocument("password", new BsonString("NEWPW"))))
+        .subscribe(
+            toSubscriber(
+                updateResult -> {
+                  result.complete(((UpdateResult) updateResult));
+                  collection
+                      .estimatedDocumentCount()
+                      .subscribe(toSubscriber(o -> count.complete(((Integer) o))));
+                }));
+    return (int) result.get(30, TimeUnit.SECONDS).getModifiedCount();
```

Comment:
Because the original groovy code had an int return type for this method, it was initially considered to remain unchanged. But I also think that using long is better than int. I wonder if others think so?

Reply:
If we keep using int as the return type, I think we can use the Math.toIntExact method to convert long to int, throwing an exception if the value overflows an int.

Reply:
@laurit We have previously discussed the adjustment of method return types here.

Reply:
@laurit I will change the return type of the involved method from int to long.